src/renderer/widgets/AssetTransactionModal/ui/AssetTransactionModal.tsx

```diff
@@ -0,0 +1,109 @@
+import { useUnit } from 'effector-react';
+import { Link } from 'react-router-dom';
+
+import { useI18n } from '@app/providers';
+import { BaseModal, BodyText, FootnoteText, HeaderTitleText, Icon, SearchInput } from '@shared/ui';
+import { useModalClose } from '@shared/lib/hooks';
+import { totalAmount } from '@shared/lib/utils';
+import { PathType, Paths, createLink } from '@shared/routes';
+import { ChainIcon } from '@entities/chain';
+import { networkModel } from '@entities/network';
+import { AssetFiatBalance } from '@entities/price';
+import { assetTransactionModel } from '../model/asset-transaction-model';
+import { assetTransactionUtils } from '../lib/utils';
+import { ModalType } from '../lib/types';
+
+type ModalDetailsProps = { title: string; path: PathType };
+
+const getModalDetails = (modalType: ModalType): ModalDetailsProps => {
+  const modalDetailsMap = {
+    [ModalType.TRANSFER]: {
+      title: 'operations.modalTitles.transferOn',
+      path: Paths.TRANSFER_ASSET,
+    },
+    [ModalType.RECEIVE]: {
+      title: 'receive.title',
+      path: Paths.RECEIVE_ASSET,
+    },
+  };
+
+  return modalDetailsMap[modalType];
+};
+
+export const AssetTransactionModal = () => {
+  const { t } = useI18n();
+
+  const assetWithChains = useUnit(assetTransactionModel.$assetWithChains);
+  const step = useUnit(assetTransactionModel.$step);
+  const modalType = useUnit(assetTransactionModel.$modalType);
+  const query = useUnit(assetTransactionModel.$query);
+  const chains = useUnit(networkModel.$chains);
+
+  const [isModalOpen, closeModal] = useModalClose(
+    !assetTransactionUtils.isNoneStep(step),
+    assetTransactionModel.output.flowClosed,
+  );
+
+  if (!assetWithChains || modalType === null) return null;
+
+  const path = getModalDetails(modalType).path;
```

Comment:
What about this one?
```suggestion
  const { title, path } = getModalDetails(modalType);
```